src/na/na_cci.c

```diff
@@ -1919,6 +1923,13 @@ na_cci_progress(na_class_t * na_class, na_context_t * context,
 		hg_time_t	t1, t2;
 		cci_event_t	*event = NULL;
 
+		struct pollfd pfd;
+
+		pfd.fd = NA_CCI_PRIVATE_DATA(na_class)->fd;
+		pfd.events = POLLIN;
+
+		poll(&pfd, 1, ((int) remaining) * 1000);
+
 		hg_time_get_current(&t1);
 
 		rc = cci_get_event(e, &event);
```

Comment:
A little further down (this patch doesn't show the line in question) there is a "remaining -= hg_time_to_double(...)" call.  There should be an if statement after that to see if remaining goes negative; if so we need to return NA_TIMEOUT right there instead of looping back around.  Otherwise the remaining variable could continue to go further and further negative.